pattern_import_export/models/ir_exports.py

```diff
@@ -16,133 +15,134 @@ class IrExports(models.Model):
-            book.close()
-            attachment_datas = base64.b64encode(pattern_file.getvalue())
-            self.env["ir.attachment"].create(
+            records = self.env[export.model_id.model].browse()
+            attachment = export._export_with_record(records)
+            export.write(
                 {
-                    "name": export.name + ".xlsx",
-                    "type": "binary",
-                    "res_id": export.id,
-                    "res_model": "ir.exports",
-                    "datas": attachment_datas,
+                    "pattern_file": attachment.datas,
+                    "pattern_last_generation_date": fields.Datetime.now(),
                 }
             )
+            attachment.unlink()
         return True
 
-
-class IrExportsLine(models.Model):
-    _inherit = "ir.exports.line"
-
-    select_tab_id = fields.Many2one("ir.exports.select.tab", string="Select tab")
-    is_many2x = fields.Boolean(
-        string="Is Many2x field", compute="_compute_is_many2x", store=True
-    )
-    related_model_id = fields.Many2one(
-        "ir.model",
-        string="Related model",
-        compute="_compute_related_model_id",
-        store=True,
-    )
-
-    def _get_last_field(self, model, path):
-        if "/" not in path:
-            path = path + "/"
-        field, path = path.split("/", 1)
-        if path:
-            model = self.env[model]._fields[field]._related_comodel_name
-            return self._get_last_field(model, path)
-        else:
-            return field, model
-
     @api.multi
-    @api.depends("name")
-    def _compute_is_many2x(self):
-        for export_line in self:
-            if export_line.export_id.resource and export_line.name:
-                field, model = export_line._get_last_field(
-                    export_line.export_id.resource, export_line.name
-                )
-                if self.env[model]._fields[field].type in ["many2one", "many2many"]:
-                    export_line.is_many2x = True
+    def _get_data_to_export(self, records):
+        """
+        Iterator who built data dict record by record
+        @param records: recordset
+        @return: dict
+        """
+        self.ensure_one()
+        headers = self._get_header()
+        for record in records:
+            data = {}
+            fields_to_export = []
+            column_headers = {}
+            for export_line in self.export_fields:
+                nb_column = 1
+                if export_line.is_many2x and export_line.select_tab_id:
+                    field_name = export_line.select_tab_id.field_id.name
+                    field = export_line.name + "/" + field_name
+                    if export_line.is_many2many:
+                        nb_column = max(1, export_line.number_occurence)
+                else:
+                    field = export_line.name
+                for _i in range(0, nb_column):
+                    fields_to_export.append(field)
```

Comment:
I think the same but it's more complicated. As the `O2M`/`M2M` could be empty, the easiest way that I found is this solution. And I also do this to avoid too much level of loops (there is already 2 embedded loop into next lines).

Reply:
I am adding a issue to discuss about it and improve it in a different PR. https://github.com/shopinvader/pattern-import-export/issues/12